vars/multicloudDeploy.groovy

```diff
@@ -114,7 +109,7 @@ void call(parameters = [:]) {
                     )
 
                 }
-                setDeployment(deployments, deployment, index, script, stageName)
+                deployment.run()
```

Comment:
I believe this change here is wrong.  Convince me of the contrary.

* Before this change: There was a closure added to map `deployments`, but nothing was really executed at that point in time.

* After this change: the closure denoted by `deployment` gets executed immediately.


I would expect something like: `deployments[SOME_INDEX_GOES_HERE]=deployment`

Beside this: the `containerMap` gets dropped. I'm not deep enough in the code for judging about that, but looks strange to me. Some explanation why this is not needed would be good.

Reply:
Yes, so far I am not aware of any use case where this is needed, so I would prefer to keep it simple if there is not a use case for that, for simple neoDeploy and cloudFoundryDeploy is not needed. I would say some explanation why this is needed would be even better. I just tried to keep compatibility, but I am not aware why it is needed. @daniel-kurzynski do you need it? for the simple use case (deploying an app to neo and cloudfoundry) I don't need it.

Reply:
Why at all a closure?

Reply:
The Closure was needed in our case, because we are have a feature in our step implementation which is capable of running the deployments in parallel. Therefore, we first build the closures and then run it in parallel. This feature is missing by the way.

Reply:
Thank you for the clarification, we were thinking about something like this, but I did check and there is no functionality in the current method (the method was already there), so I decided to remove it.